File: ptr-lib/src/main/java/wtf/s1/android/ptr/PtrListener.java
package wtf.s1.android.ptr;

import wtf.s1.android.ptr.indicator.PtrIndicator;

/**
 *
 */
public interface PtrListener {

    /**
     * When the content view has reached top and refresh has been completed, view will be reset.
     *
     * @param frame
     */
    public void onReset(PtrFrameLayout frame);

    /**
     * prepare for loading
     *
     * @param frame
     */
    public void onPrepare(PtrFrameLayout frame);

    /**
     * perform refreshing UI
     */
    public void onBegin(PtrFrameLayout frame);

    /**
     * perform UI after refresh
     */
    public void onComplete(PtrFrameLayout frame);


    public void onPositionChange(PtrFrameLayout frame, int status, PtrIndicator ptrIndicator);
}

File: ptr-lib/src/main/java/wtf/s1/android/ptr/PtrListener.kt
package wtf.s1.android.ptr

/**
 *
 */
interface PtrListener {

    /**
     * prepare for loading
     *
     * @param frame
     */
    fun onDrag(frame: PtrLayout) {}

    /**
     * perform refreshing UI
     */
    fun onRefreshing(frame: PtrLayout) {}

    /**
     * perform UI after refresh
     */
    fun onComplete(frame: PtrLayout) {}

    fun onPositionChange(frame: PtrLayout) {}
}